d2/src/main/java/com/linkedin/d2/balancer/strategies/relative/StateUpdater.java

```diff
@@ -167,12 +167,18 @@ int getFirstValidPartitionId()
    */
   void updateState()
   {
-    // Update state for each partition
-    for (Integer partitionId : _partitionLoadBalancerStateMap.keySet())
+    try {
+      // Update state for each partition
+      for (Integer partitionId : _partitionLoadBalancerStateMap.keySet())
+      {
+        PartitionState partitionState = _partitionLoadBalancerStateMap.get(partitionId);
+        updateStateForPartition(partitionState.getTrackerClients(), partitionId, partitionState, partitionState.getClusterGenerationId());
+      }
+    } catch (Exception ex)
     {
-      PartitionState partitionState = _partitionLoadBalancerStateMap.get(partitionId);
-      updateStateForPartition(partitionState.getTrackerClients(), partitionId, partitionState, partitionState.getClusterGenerationId());
+      LOG.error("Failed update the event.", ex);
```

Comment:
Maybe `"Failed to update the state for service: " + _serviceName`?

Reply:
Fixed